src/System.Windows.Forms/src/System/Windows/Forms/TrackBar.cs

```diff
@@ -557,11 +566,26 @@ public int TickFrequency
                 }
 
                 _tickFrequency = value;
-                if (IsHandleCreated)
+                // set the recreateHandle flag to null,
+                // if the _autoDrawTicks flag does not equal what ShouldAutoDrawTicks()
+                // we set the recreateHandle to true,
+                // then we skip sending the messages and,
+                // recreateHandle will be called at the end of the method.
+                bool recreateHandle = false;
+                if (IsHandleCreated && _autoDrawTicks != ShouldAutoDrawTicks())
                 {
-                    PInvoke.SendMessage(this, (User32.WM)PInvoke.TBM_SETTICFREQ, (WPARAM)value);
+                    recreateHandle = true;
+                }
```

Comment:
Since we use the identical code in two places, perhaps it makes sense to extract it into a new method, e.g.:
```cs
        /// <summary>
        ///  Indicates if the native control needs to be recreated because the decision of whether the ticks drawing is
        ///  performed by the native control or the Windows Forms runtime is no longer valid.
        /// </summary>
        private bool ShouldRecreateHandle() => IsHandleCreated && _autoDrawTicks != ShouldAutoDrawTicks();
```

```suggestion
                bool recreateHandle = ShouldRecreateHandle();
```